src/lib/hooks/useTokenList.ts

```diff
@@ -0,0 +1,69 @@
+import { TokenInfo, TokenList } from '@uniswap/token-lists'
+import { atom, useAtom } from 'jotai'
+import useActiveWeb3React from 'lib/hooks/useActiveWeb3React'
+import fetchTokenList, { getTokenInfo } from 'lib/utils/fetchTokenList'
+import resolveENSContentHash from 'lib/utils/resolveENSContentHash'
+import { useEffect, useMemo, useState } from 'react'
+import { WrappedTokenInfo } from 'state/lists/wrappedTokenInfo'
+
+export type TokenMap = Readonly<{ [tokenAddress: string]: WrappedTokenInfo }>
+type ChainTokenMap = Readonly<{ [chainId: number]: TokenMap }>
+type Mutable<T> = {
+  -readonly [P in keyof T]: Mutable<T[P]>
+}
+
+const chainTokenMapAtom = atom<ChainTokenMap>({})
+
+function tokensToTokenMap(tokens: TokenList | TokenInfo[]) {
```

Comment:
For cases like this where we're duping code (sorry if there's a diff I'm missing) I think maybe we should create a new file in src/hooks or wherever it is currently, and import it in both its prior and new use locations, instead of duping.

Reply:
yeah agreed we should just re-use in a shared file